src/Symfony/Component/HttpFoundation/File/File.php

```diff
@@ -61,6 +61,26 @@ public function guessExtension()
         return $guesser->guess($type);
     }
 
+    /**
+     * Returns the extension based on the original name with pathinfo.
+     *
+     * This method uses pathinfo() function.
+     *
+     * @return string|null The pathinfo extension or null if it cannot be guessed
+     *
+     * @see pathinfo()
+     */
+    public function getPathinfoExtension()
+    {
+        $extension = null;
+        $info = pathinfo($this->getClientOriginalName());
```

Comment:
The `getClientOriginalName()` method is only available in the `UploadedFile` class